lib/excision_web/controllers/classifier_controller.ex

```
@@ -192,34 +196,48 @@ defmodule ExcisionWeb.ClassifierController do
       case Plug.Conn.get_resp_header(resp, "content-encoding") do
         ["gzip"] ->
-          :zlib.gunzip(resp.resp_body)
+          {:ok, :zlib.gunzip(resp.resp_body)}
 
         ["br"] ->
-          {:ok, data} = :brotli.decode(resp.resp_body)
-          data
+          :brotli.decode(resp.resp_body)
 
         _ ->
-          resp.resp_body
+          if resp.status >= 400 do
+            Logger.error(
+              "Got failure response while proxying request for classifier #{classifier.name}: #{resp.status} #{resp.resp_body}"
+            )
+
+            {:error, resp.resp_body}
+          else
+            {:ok, resp.resp_body}
+          end
       end
-      |> Jason.decode!()
 
-    # record the decision
-    Excision.Excisions.create_decision(%{
-      decision_site_id: decision_site.id,
-      classifier_id: classifier.id,
-      input: Jason.encode!(req_body["messages"]),
-      prediction_id:
-        decision_site.choices
-        |> Enum.find(fn c ->
-          c.name ==
-            resp_body["choices"]
-            |> hd()
-            |> then(& &1["message"]["content"])
-            |> then(&Jason.decode!/1)
-            |> then(& &1["value"])
-        end)
-        |> then(& &1.id)
-    })
+    deserialized_body = resp_body |> elem(1) |> Jason.decode!()
+
+    case resp_body do
+      {:ok, _} ->
+        # record the decision
+        Excision.Excisions.create_decision(%{
+          decision_site_id: decision_site.id,
+          classifier_id: classifier.id,
+          input: Jason.encode!(req_body["messages"]),
+          prediction_id:
+            decision_site.choices
+            |> Enum.find(fn c ->
+              c.name ==
+                deserialized_body["choices"]
+                |> hd()
+                |> then(& &1["message"]["content"])
+                |> then(&Jason.decode!/1)
+                |> then(& &1["value"])
+            end)
+            |> then(& &1.id)
+        })
+
+      {:error, _} ->
+        nil
```

Can we return a useful error to the end user?